src/builtin/filter/datetime.rs

```diff
@@ -4,17 +4,29 @@ use chrono::{DateTime, NaiveDateTime, Utc};
 use serde_json::Value;
 
 use crate::context::Context;
-use crate::error::Result;
-
-fn format_timestamp(filter: &str, input: &Value, args: &HashMap<String, Value>, default: &str) -> Result<Value> {
-    let ts = input
-        .as_i64()
-        .ok_or_else(|| format!("`{}` accepts integer only", filter))?;
+use crate::error::*;
+
+fn format_timestamp(
+    filter: &'static str,
+    input: &Value,
+    args: &HashMap<String, Value>,
+    default: &str,
+) -> Result<Value> {
+    let ts = input.as_i64().ok_or_else(|| {
+        Error::with_message("invalid input type")
+            .context("filter", filter)
+            .context("expected", "i64")
+            .context("input", input.to_string())
+    })?;
 
     let format = match args.get("format") {
-        Some(x) => x
-            .as_str()
-            .ok_or_else(|| format!("`{}` format must be a string", filter))?,
+        Some(x) => x.as_str().ok_or_else(|| {
+            Error::with_message("invalid argument type")
+                .context("filter", filter)
```

Comment:
I like this, ability to add context fluently

Reply:
On some other note - I think (later?) it could help to showcase how cool this is, with some doc-tests maybe ? It may not be that easy to discover how powerful the usage can be - e.g. for people that would need to add more error edge cases handlingin the future but not necessarily know about this.

Reply:
@cyplo What kind of people do you mean? I mean, this is an error for the user. How user can add more edge cases handling? I probably don't understand what you do mean.

Reply:
yeah, sorry for not being precise here. 
I meant future maintainers of this library, temen. Either me and you in the future, not remembering how to do that, or some other people that are fresh to this codebase.

Reply:
@cyplo Hmm, usage of this simple error should be crystal clear, that's the reason why docs are generated. Not sure if it's worth it, but I did add an example to the `error` module.

<img width="1012" alt="screenshot 2018-11-16 at 10 17 08" src="https://user-images.githubusercontent.com/1084172/48612320-f7fa4c00-e988-11e8-9ef2-3a84b4453287.png">